File: kltn-backend/src/main/java/iuh/fit/se/dtos/responses/CalendarEventResponse.java
package iuh.fit.se.dtos.responses;

import com.fasterxml.jackson.annotation.JsonFormat;
import iuh.fit.se.enums.EventPriority;
import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class CalendarEventResponse {
  private Long id;
  private String title;
  private String description;

  @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
  private LocalDateTime startTime;

  @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
  private LocalDateTime endTime;

  private Boolean allDay;
  private EventPriority priority;

  private Long createdById;
  private String createdByUsername;
}

File: kltn-frontend-admin/src/types/responses/calendarEventResponse.ts

export type EventPriority = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';

export interface CalendarEventResponse {
  id: number;
  title: string;
  description?: string;
  startTime: string; // ISO
  endTime: string; // ISO
  allDay?: boolean;
  priority?: EventPriority;
  createdById?: number;
  createdByUsername?: string;
}
